flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingMetricEvaluator.java

```diff
@@ -296,8 +297,14 @@ protected static void computeProcessingRateThresholds(
             upperUtilization = 1.0;
             lowerUtilization = 0.0;
         } else {
-            upperUtilization = targetUtilization + utilizationBoundary;
-            lowerUtilization = targetUtilization - utilizationBoundary;
+            upperUtilization =
+                    conf.getOptional(TARGET_UTILIZATION_BOUNDARY)
+                            .map(boundary -> targetUtilization + boundary)
+                            .orElseGet(() -> conf.get(UTILIZATION_MAX));
+            lowerUtilization =
+                    conf.getOptional(TARGET_UTILIZATION_BOUNDARY)
+                            .map(boundary -> targetUtilization - boundary)
+                            .orElseGet(() -> conf.get(UTILIZATION_MIN));
```

Comment:
IIUC, the compatibility logic is not as expected. `TARGET_UTILIZATION_BOUNDARY` should be only used when  `UTILIZATION_MAX` or `UTILIZATION_MIN` are not set by users.

If `UTILIZATION_MAX` or `UTILIZATION_MIN` is set by users, it means users are using the latest options. So we should ignore `TARGET_UTILIZATION_BOUNDARY` even if it's set as well. WDYT?

Also, it should be tested as well.

Reply:
fixed.